<a href="https://colab.research.google.com/github/nurfnick/Numerical_Methods/blob/master/Module_6_Regression/WeBWorK6HomeworkHelper.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# WeBWorK Homework Helper 6  Regression



##WeBWorK 6

Is the number of games won by a major league baseball team in a season related to the team batting average? The table below shows the number of games won and the batting average (in thousandths) of 8 teams.


| Team | Games Won | Batting Average |
|------|-----------|-----------------|
| 1    | 64        | 267             |
| 2    | 61        | 261             |
| 3    | 76        | 279             |
| 4    | 71        | 288             |
| 5    | 97        | 276             |
| 6    | 65        | 278             |
| 7    | 113       | 269             |
| 8    | 120       | 275             |


Using games won as the explanatory variable $x$ find the correlation coefficient and the equation of the least squares line.

In [ ]:
from scipy import stats

games = [64,61,76,71,97,65,113,120]
BA = [267, 261,279,288,276,278,269,275]

stats.linregress(games, BA)

LinregressResult(slope=0.015323292057372496, intercept=272.84742052471654, rvalue=0.04304300377517962, pvalue=0.9193939947428843, stderr=0.1452015328046192)

So the correlations coefficient is $r = 0.04304300377517962$ and the equation for the line is $\hat y = 0.015323292057372496 x +272.84742052471654$


## WeBWorK 7

Calculate the least-squares regression line for the following set of data points:
(-3,6), (2,5), (2,6), (1,7), (6,9), (4,7), (9,9).

I am going to answer this one the hard way, using the forumlas!

$$
S_{xy} = \sum x_iy_i-n\bar x\bar y\\
S_{xx} = \sum x_i^2 - n\bar x^2\\
a_1 = \frac{S_{xy}}{S_{xx}}\\
a_0 = \bar y - a_1 \bar x
$$

In [2]:
x =[-3,2,2,1,6,4,9]
y = [6,5,6,7,9,7,9]
n = len(x)

xbar = sum(x)/n
ybar = sum(y)/n
xy = [x[i]*y[i] for i in range(n)]
xx = [x**2 for x in x]

sxy = sum(xy) - n*xbar*ybar
sxx = sum(xx) - n*xbar**2

a1 = sxy/sxx
a0 = ybar - a1*xbar

print(a0,a1)

6.079545454545455 0.3068181818181818


## WeBWorK 11

The amounts of 6 restaurant bills and the corresponding amounts of the tips are given in the below.
\begin{array}{c|cccccc}
\mbox{Bill} & 70.29 & 106.27 & 32.98 & 64.30 & 52.44 & 49.72 \cr
\hline
\mbox{Tip}  & 10.00 & 16.00 & 4.50 & 7.70 & 7.00 & 5.28 \cr
\end{array}
Use a 0.05 confidence level to find the following:
The test statistic r = 

Is there a significant correlation?

 1. Yes
 1. No

If the amount of the bill is $40, the best prediction for the amount of the tip is 


In [ ]:
bill = [70.29,106.27,32.98,64.30,52.44,49.72]
tip = [10,16,4.50,7.7,7,5.28]

stats.linregress(bill, tip)

LinregressResult(slope=0.16463581370171665, intercept=-1.9038443253075759, rvalue=0.9812002855581488, pvalue=0.0005268217100319339, stderr=0.01619110428907605)

Yes significant correlation, the $p$ value is low!

As for making the prediction.

In [ ]:
def f(x):
  return x*0.16463581370171665-1.9038443253075759

In [ ]:
f(40)

4.68158822276109